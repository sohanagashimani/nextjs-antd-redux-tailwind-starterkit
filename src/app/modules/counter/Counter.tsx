'use client';
import { Button } from 'antd';

import Image from 'next/image';

import { useAppDispatch } from '@/hooks/useAppDispatch';
import { useAppSelector } from '@/hooks/useAppSelector';
import { decrementCount, incrementCount } from '@/redux/slices/counterSlice';

export default function Counter() {
  const count = useAppSelector(state => state.counter.count);
  const dispatch = useAppDispatch();

  const handleIncrement = () => {
    dispatch(incrementCount());
  };

  const handleDecrement = () => {
    dispatch(decrementCount());
  };
  return (
    <main className="flex min-h-screen flex-col items-center justify-center gap-6 bg-black text-white">
      <Image
        className="dark:invert"
        src="/next.svg"
        alt="Next.js Logo"
        width={180}
        height={37}
        priority
      />
      <h1 className="text-5xl font-medium">Counter</h1>
      <div className="flex items-center">
        <Button
          type="primary"
          data-testid="subtract-button"
          onClick={handleDecrement}
          className="p-4"
        >
          -
        </Button>
        <div data-testid="count" className="p-4 text-2xl">
          {count}
        </div>
        <Button
          type="primary"
          onClick={handleIncrement}
          data-testid="add-button"
          className="p-4"
        >
          +
        </Button>
      </div>
    </main>
  );
}
