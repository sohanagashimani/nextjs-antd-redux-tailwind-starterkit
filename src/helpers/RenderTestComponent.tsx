import { render } from '@testing-library/react';
import { ReactNode } from 'react';

import { StoreProvider } from '@/redux/StoreProvider';

const RenderTestComponent = (children: ReactNode) => {
  return render(<StoreProvider>{children}</StoreProvider>);
};
export default RenderTestComponent;
