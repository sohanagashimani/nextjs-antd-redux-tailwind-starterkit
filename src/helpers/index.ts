export { default as RenderTestComponent } from './RenderTestComponent';
