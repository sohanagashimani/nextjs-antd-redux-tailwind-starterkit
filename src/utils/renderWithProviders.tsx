import { RenderOptions, render } from '@testing-library/react';
import React, { PropsWithChildren } from 'react';
import { Provider } from 'react-redux';

import { AppStore, RootState, setupStore } from '@/redux/store';

// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: AppStore;
}

const renderWithProviders = (
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) => {
  const Wrapper = ({ children }: PropsWithChildren) => {
    return <Provider store={store}>{children}</Provider>;
  };

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export default renderWithProviders;
