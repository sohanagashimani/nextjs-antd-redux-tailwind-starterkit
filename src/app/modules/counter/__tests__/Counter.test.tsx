import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';

import { act } from 'react-dom/test-utils';

import Counter from '@/app/modules/counter/Counter';
import { RenderTestComponent } from '@/helpers';
import renderWithProviders from '@/utils/renderWithProviders';

describe('Counter.tsx', () => {
  it('Test increment count', () => {
    renderWithProviders(<Counter />);
    const addButton = screen.getByTestId('add-button');
    const count = screen.getByTestId('count');
    expect(count).toHaveTextContent('0');

    act(() => {
      addButton.click();
    });
    expect(count).toHaveTextContent('1');
  });

  it('Test decrement count', () => {
    renderWithProviders(<Counter />, {
      preloadedState: { counter: { count: 1 } },
    });
    const subButton = screen.getByTestId('subtract-button');
    const count = screen.getByTestId('count');
    act(() => {
      subButton.click();
    });

    expect(count).toHaveTextContent('0');
  });

  it('snapshot testing', () => {
    const { container } = RenderTestComponent(<Counter />);
    expect(container).toMatchSnapshot();
  });
});
