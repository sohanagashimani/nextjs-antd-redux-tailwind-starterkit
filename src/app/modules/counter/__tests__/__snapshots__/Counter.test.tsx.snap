// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.tsx snapshot testing 1`] = `
<div>
  <main
    class="flex min-h-screen flex-col items-center justify-center gap-6 bg-black text-white"
  >
    <img
      alt="Next.js Logo"
      class="dark:invert"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="37"
      src="/next.svg"
      style="color: transparent;"
      width="180"
    />
    <h1
      class="text-5xl font-medium"
    >
      Counter
    </h1>
    <div
      class="flex items-center"
    >
      <button
        class="ant-btn css-dev-only-do-not-override-1b0bdye ant-btn-primary p-4"
        data-testid="subtract-button"
        type="button"
      >
        <span>
          -
        </span>
      </button>
      <div
        class="p-4 text-2xl"
        data-testid="count"
      >
        0
      </div>
      <button
        class="ant-btn css-dev-only-do-not-override-1b0bdye ant-btn-primary p-4"
        data-testid="add-button"
        type="button"
      >
        <span>
          +
        </span>
      </button>
    </div>
  </main>
</div>
`;
